vm/src/types/program.rs

```diff
@@ -40,25 +41,64 @@ use std::path::Path;
 // exceptional circumstances, such as when reconstructing a backtrace on execution
 // failures.
 // Fields in `Program` (other than `SharedProgramData` itself) are used by the main logic.
-#[derive(Clone, Default, Debug, PartialEq, Eq)]
-pub(crate) struct SharedProgramData {
-    pub(crate) data: Vec<MaybeRelocatable>,
-    pub(crate) hints: HashMap<usize, Vec<HintParams>>,
-    pub(crate) main: Option<usize>,
+#[derive(Clone, Default, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SharedProgramData {
+    pub data: Vec<MaybeRelocatable>,
+    pub hints: HashMap<usize, Vec<HintParams>>,
+    pub main: Option<usize>,
     //start and end labels will only be used in proof-mode
-    pub(crate) start: Option<usize>,
-    pub(crate) end: Option<usize>,
-    pub(crate) error_message_attributes: Vec<Attribute>,
-    pub(crate) instruction_locations: Option<HashMap<usize, InstructionLocation>>,
-    pub(crate) identifiers: HashMap<String, Identifier>,
-    pub(crate) reference_manager: Vec<HintReference>,
+    pub start: Option<usize>,
+    pub end: Option<usize>,
+    pub error_message_attributes: Vec<Attribute>,
+    pub instruction_locations: Option<HashMap<usize, InstructionLocation>>,
+    pub identifiers: HashMap<String, Identifier>,
+    pub reference_manager: Vec<HintReference>,
 }
 
+
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Program {
```

Comment:
Can't you derive Serialize and Deserialize on Program?  Maybe it's a little tricky with the Arc, not sure, but I think if only the Arc is the issue it would be cleaner to use `#[serde(with = "module")]` only on the Arc then and implement serde for it

Reply:
resolved via [this](https://github.com/kkrt-labs/cairo-vm/pull/2/commits/11fe551e5ea078d13e275524dcbbcb174c4ec21d) commit.